### PR TITLE
feat(router): add /relay endpoint

### PR DESCRIPTION
**Pull request by ShankarSinghC**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

/relay is a api that will be used to just forward the request sent my merchant without performing any application logic on it. In this pr the changes is particular to support the refunds for the payments that was directly performed with the connector (example stripe) with out hyperswitch involved during the payment flow.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

image

image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: api-reference/api-reference/relay/relay-retrieve.mdx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+openapi: openapi_spec get /relay/{relay_id}
```
Comment (sai-harsha-vardhan) [inline api-reference/api-reference/relay/relay-retrieve.mdx:2]: we'll have to make changes in `mint.json` also, we can test the api-reference changes locally using mintlify - https://mintlify.com/docs/development

File: crates/api_models/src/relay.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::enums;
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+pub struct RelayRequest {
+    /// The identifier that is associated to a resource at the connector reference to which the relay request is being made
+    pub connector_resource_id: String,
+    /// Identifier of the connector ( merchant connector account ) which was chosen to make the payment
```
Comment (sai-harsha-vardhan) [inline crates/api_models/src/relay.rs:10]: Not only for a payment right?

File: crates/api_models/src/relay.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    #[schema(example = "mca_5apGeP94tMts6rg3U3kR", value_type = String)]
+    pub connector_id: common_utils::id_type::MerchantConnectorAccountId,
+    /// The business profile that is associated with this relay request.
+    #[schema(example = "pro_abcdefghijklmnopqrstuvwxyz", value_type = String)]
+    pub profile_id: common_utils::id_type::ProfileId,
+    /// The type of relay request
+    #[serde(rename = "type")]
+    pub relay_type: RelayType,
+    /// The data that is associated with the relay request
+    pub data: Option<RelayData>,
```
Comment (sai-harsha-vardhan) [inline crates/api_models/src/relay.rs:74]: Please add `connector_reference_id` in the response

File: crates/api_models/src/relay.rs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+pub struct RelayRequest {
+    /// The identifier that is associated to a resource at the connector reference to which the relay request is being made
+    #[schema(example = "7256228702616471803954")]
+    pub connector_resource_id: String,
+    /// Identifier of the connector ( merchant connector account ) to which relay request is being made
+    #[schema(example = "mca_5apGeP94tMts6rg3U3kR", value_type = String)]
+    pub connector_id: common_utils::id_type::MerchantConnectorAccountId,
+    /// The business profile that is associated with this payment request
```
Comment (sai-harsha-vardhan) [inline crates/api_models/src/relay.rs:15]:
```suggestion
    /// The business profile that is associated with this relay request
```

**Review by sai-harsha-vardhan — DISMISSED** [review]

File: crates/api_models/src/relay.rs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+pub struct RelayRequest {
+    /// The identifier that is associated to a resource at the connector to which the relay request is being made
+    #[schema(example = "7256228702616471803954")]
+    pub connector_resource_id: String,
+    /// Identifier of the connector ( merchant connector account ) to which relay request is being made
+    #[schema(example = "mca_5apGeP94tMts6rg3U3kR", value_type = String)]
+    pub connector_id: common_utils::id_type::MerchantConnectorAccountId,
+    /// The business profile that is associated with this relay request
+    #[schema(example = "pro_abcdefghijklmnopqrstuvwxyz", value_type = String)]
+    pub profile_id: common_utils::id_type::ProfileId,
```
Comment (jarnura) [inline crates/api_models/src/relay.rs:17]: move this to headers

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

File: crates/api_models/src/relay.rs [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RelayType {
+    /// The relay request is for a refund
+    Refund,
+}
+
+#[derive(Debug, ToSchema, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case", untagged)]
```
Comment (Narayanbhat166) [inline crates/api_models/src/relay.rs:30]: do not use untagged for deserialization. You can rely on the `relay_type` to infer this. Can be changed in the subsequent PRs as well

**Pull request merged**